Doc/library/typing.rst

```diff
@@ -849,6 +849,18 @@ using ``[]``.
       concat(b"foo", b"bar")  # OK, output has type 'bytes'
       concat("foo", b"bar")   # Error, cannot mix str and bytes
 
+   Note that, despite its name, ``AnyStr`` has nothing to do with the
+   :class:`Any` type, nor does it mean "any string". In particular, ``AnyStr``
+   and ``str | bytes`` are different from each other and have different use
+   cases. If ``AnyStr`` is not part of the input arguments it should not be
+   used::
+
+      def greet_bad(cond: bool) -> AnyStr:
+          return "hi there!" if cond else b"greetings!"  # Error
```

Comment:
Unfortunately, I believe not all type checkers will _necessarily_ report an error here, even though it's an invalid, and incorrect, use of `AnyStr`. Maybe:

```suggestion
      # Invalid use of AnyStr:
      # The type variable is used only once in the function signature,
      # so cannot be "solved" by the type checker
      def greet_bad(cond: bool) -> AnyStr:
          return "hi there!" if cond else b"greetings!"
```

Reply:
I like this wording! Will use it as-is.